tests/Aql/AqlBuildSupervisedAqlValueTest.cpp

```diff
@@ -0,0 +1,196 @@
+#include "gtest/gtest.h"
+#include "Basics/ResourceUsage.h"
+#include "Basics/SupervisedBuffer.h"
+#include "Aql/AqlValue.h"
+#include <velocypack/Builder.h>
+#include <string>
+
+using arangodb::GlobalResourceMonitor;
+using arangodb::ResourceMonitor;
+using arangodb::ResourceUsageScope;
+using arangodb::aql::AqlValue;
+using arangodb::aql::buildSupervisedAqlValue;
+using arangodb::basics::SupervisedBuffer;
+using arangodb::velocypack::Builder;
+using arangodb::velocypack::Value;
+
+TEST(BuildSupervisedAqlValueTest,
+     AccountsMemoryLargeAndSmallValuesNormalBuffer) {
+  ResourceMonitor monitor(GlobalResourceMonitor::instance());
+
+  {
+    ResourceUsageScope usageScope(monitor);
+    AqlValue largeValue;
+    {
+      Builder builder;
+      builder.openArray();
+      builder.add(Value(std::string(1024, 'a')));
+      builder.close();
+
+      ASSERT_EQ(monitor.current(), 0);
```

Comment:
This is unexpected to me!

We are writing something into the builder, why is it not consuming memory?